doc/jetty-config.md

```diff
@@ -16,21 +16,60 @@ This sets what port to use for _unencrypted_ HTTP traffic.  If not supplied, but
 `host` is supplied, a value of 8080 will be used.  If neither host nor port is
 supplied, we won't listen for unencrypted traffic at all.
 
+### `acceptor-threads`
+
+This sets the number of threads that the webserver will dedicate to accepting
+socket connections for _unencrypted_ HTTP traffic.  Defaults to the number of
+virtual cores on the host divided by 8, with a minimum of 1 and maximum of 4.
+
+### `selector-threads`
+
+This sets the number of threads that the webserver will dedicate to processing
+events on connected sockets for _unencrypted_ HTTP traffic.  Defaults to the
+number of virtual cores on the host divided by 2, with a minimum of 1 and
+maximum of 4.
+
 ### `max-threads`
 
-This sets the maximum number of threads assigned to responding to HTTP and HTTPS
-requests, effectively changing how many concurrent requests can be made at one
-time. Defaults to 100.
+This sets the maximum number of threads assigned to responding to HTTP and/or
+HTTPS requests for a single webserver, effectively changing how many concurrent
+requests can be made at one time.  Defaults to 200.
+
+Each webserver instance requires a minimum number of threads in order to boot
+properly.  The minimum number is calculated as:
+
+~~~~
+(number of "acceptor-threads" for each port) +
+(number of "selector-threads" for each port) +
+1 "worker" thread
+~~~~
+
+"1" is the minimum number of worker threads required to process incoming
+web requests.
+
+For example, if an _unencrypted_ port with 2 `acceptor-threads` and 3
+`selector-threads` and an _encrypted_ port with 4 `acceptor-threads` and 5
+`selector-threads` were configured with the webserver, the webserver would
+require that a minimum value of 15 (2 + 3 + 4 + 5 + 1) be used for the
+`max-threads` setting.  If the configured value for `max-threads` is less
+than the minimum required value, server startup will fail with an
+`IllegalStateException`, with a message containing the words
+"insufficient threads".
+
+Note that each web request must be processed on a "worker" thread which is
+separate from the acceptor and selector threads.  The configured `max-threads`
+value should allow for the maximum number of requests which are desired for the
+server be able to handle concurrently.
```

Comment:
Worker threads effectively are configurable.  Basically, the difference between the sums of `acceptor-threads` and `selector-threads` across connectors and `max-threads` is the maximum number of worker threads that the server will allocate.  Worker threads are different from `acceptor-threads` and `selector-threads` in they aren't fixed at boot time.  The server will only allocate one by default but can allocate additional threads, as needed to handle incoming traffic, up to the `max-threads` for the pool.  The server can also let the number of worker threads scale back down after a peak period for requests has passed.

If we hadn't already had the concept of a `max-threads` setting, it might have made more sense to me to have a `worker-threads` setting and we could use that to calculate the `max-threads` that we give to Jetty during configuration.  It might be too confusing to change that now, and it also would seem to get us into the mode of having our production code be more knowledgeable about exactly how this algorithm is used by Jetty - which I think we're trying to avoid as much as possible now.